### PR TITLE
plat-stm32mp1: expose SCMI voltage domains for some PMIC regulators

### PR DESCRIPTION
**Pull request by etienne-lms**
Defines the DT binding ID and SCMI resources for SCMI voltage domain exposed by platform stm32mp1 when secure hardeing of the resources are enabled.

### CODE REVIEW

File: core/include/drivers/stpmic1_regulator.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2021, STMicroelectronics
+ */
+
+#ifndef STPMIC1_REGULATOR_H
```
Comment (jenswi-linaro) [inline core/include/drivers/stpmic1_regulator.h:6]: __DRIVERS_STPMIC1_REGULATOR_H?

File: core/include/drivers/stpmic1_regulator.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2021, STMicroelectronics
```
Comment (jenswi-linaro) [inline core/include/drivers/stpmic1_regulator.h:3]: 2022 also?

File: core/include/drivers/stpmic1_regulator.h (hunk 1 above)
Comment (etienne-lms) [reply core/include/drivers/stpmic1_regulator.h:6]: right.

**Comment by jenswi-linaro** [comment]
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**